lib/entities/quote/DutchQuote.ts

```diff
@@ -332,6 +385,22 @@ export class DutchQuote implements IQuote {
     }
   }
 
+  public get portionAmountOutStart(): BigNumber {
+    return this.amountOutStart.mul(this.portionBips ?? 0).div(BPS);
+  }
+
+  public get portionAmountOutEnd(): BigNumber {
+    return this.amountOutEnd.mul(this.portionBips ?? 0).div(BPS);
+  }
+
+  public get amountInGasAndPortionAdjusted(): BigNumber {
+    return this.amountIn.add(this.portionAmountOutStart);
+  }
+
+  public get amountOutGasAndPortionAdjusted(): BigNumber {
+    return this.amountOut.sub(this.portionAmountOutStart);
```

Comment:
I think this is also wrong; this is true for EXACT_IN but not for exactOut. But i guess for exactOut it does not really matter, because we are only interested in amountIn gasAdjusted?

Reply:
so shall we be calculating this only for EXACT_IN?